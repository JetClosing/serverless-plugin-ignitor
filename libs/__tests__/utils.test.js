const utils = require('../utils');
const path = require('path');
const fs = require('fs');

const MOCK_FILE = path.resolve(__dirname, 'example.js');

const MOCK_DIR = path.resolve(__dirname, 'foo');

test('functions exported', () => {
  expect(utils.write).toBeDefined();
  expect(utils.mkdir).toBeDefined();
  expect(utils.rm).toBeDefined();
  expect(utils.cli).toBeDefined();
});

test('write file', () => {
  utils.write(MOCK_FILE, 'console.log(\'hi\')');
  expect(fs.readFileSync(MOCK_FILE, 'utf8')).toMatchSnapshot();

  utils.rm(MOCK_FILE);
});

test('create directory', () => {
  utils.mkdir(MOCK_DIR);
  expect(fs.existsSync(MOCK_DIR)).toBeTruthy();

  utils.rm(MOCK_DIR);
});

test('test cli', () => {
  const echoResponse = utils.cli('echo "Hello"').toString('utf8');
  expect(echoResponse.trim()).toEqual('Hello');
}); 