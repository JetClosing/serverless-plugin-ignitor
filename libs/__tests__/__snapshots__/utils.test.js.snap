// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write file 1`] = `"console.log('hi')"`;
